pkg/addonmanager/manager.go

```diff
@@ -171,8 +177,15 @@ func (a *addonManager) Start(ctx context.Context) error {
 		a.addonAgents,
 	)
 
-	var addonConfigController factory.Controller
-	var managementAddonConfigController factory.Controller
+	// This is a duplicate controller in general addon-manager. This should be removed when we
+	// alway enable the addon-manager
+	addonStatusController := addonstatus.NewAddonStatusController(
```

Comment:
Does this status controller also need to be enabled by annotation?

Reply:
I think it is unnecessary, we can start this with noop. enable by annotation is weird, because user actually can add/remove annotation, and we have to watch the resource and dynamically start/stop this controller